sherpa/data.py

```diff
@@ -1353,12 +1396,77 @@ def get_xerr(self, filter=False, model=None):
         xlo, xhi = indep
         return xhi - xlo
 
+    def get_filter(self, format='%.4f', delim=':'):
+        """Return the data filter as a string.
+
+        For each noticed range the filter is reported as
+        starting at the low ege of the first bin and ends
+        at the upper edge of the last bin in the range.
+
+        .. versionchanged:: 4.14.0
+           Prior to 4.14.0 the filter used the mid-point of the bin,
+           not it's low or high value.
+
+        Parameters
+        ----------
+        format : str, optional
+            The formatting of the numeric values.
+        delim : str, optional
+            The string used to mark the low-to-high range.
+
+        See Also
+        --------
+        get_filter_expr, ignore, notice
+
+        Examples
+        --------
+
+        >>> xlo = np.asarray([1, 2, 3, 5, 6])
+        >>> xhi = xlo + 1
+        >>> y = np.ones(5)
+        >>> d = Data1DInt('example', xlo, xhi, y)
+        >>> d.get_filter()
+        '1.0000:7.0000'
+        >>> d.ignore(2.5, 4.5)
+        >>> d.get_filter()
+        '1.0000:2.0000,5.0000:7.0000'
+
+        >>> d.get_filter(format='%i', delim='-')
+        '1-2,5-7'
+
+        """
+
+        # We could just use the middle of each bin (which is what
+        # would happen if we didn't over-ride Data1D.filter) but let's
+        # use the start and end values for each selected bin.
+        #
+        indep = self.get_evaluation_indep(filter=True)
+        if len(indep) == 1:
```

Comment:
This is an operational check, but I wonder if we need a more formalized way to determine this. It turns out to be a not-insignificant thing to look at ( see #1220 ) so it's not going to be done in this PR.

Reply:
(note: for this routine we require `indep` to be a two-element sequence in line 1455 so checking the length here does make sense)